app/components/submission/header.js

```diff
@@ -393,7 +393,15 @@ export default class SubmissionHeaderComponent extends Component {
   );
 
   takeInTreatment = async () => {
-    // TODO update submission data? It could have been changed on subcase
+    const subcase = await this.args.submission.subcase;
+    if (subcase) {
```

Comment:
checking `subcase?.id` here is a little bit better, since a Promise would also be truthy.